main.nf

```diff
@@ -0,0 +1,517 @@
+Channel
+    .fromPath("$root/**/*t1.nii.gz",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .set{in_t1}
+
+Channel
+    .fromPath("$root/**/*$params.labels_img_prefix*labels.nii.gz",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .set{in_labels}
+
+in_t1
+    .join(in_labels)
+    .set{in_t1_labels}
+
+in_transfo = Channel
+    .fromFilePairs("$root/**/{*0GenericAffine.mat,*1Warp.nii.gz}",
+                    size: 2,
+                    maxDepth:1,
+                    flat: true) {it.parent.name}
+
+in_tracking = Channel
+    .fromFilePairs("$root/**/{*tracking*.*,}",
+                    size: -1,
+                    maxDepth:1) {it.parent.name}
+
+Channel
+    .fromPath("$root/**/*fodf.nii.gz",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .set{fodf_for_afd_rd}
+
+Channel.fromPath(file(params.template))
+    .into{template_for_registration;template_for_transformation_data;template_for_transformation_metrics}
+
+Channel.fromPath(file(params.labels_list))
+    .into{labels_list_for_compute;labels_list_for_visualize}
+
+in_opt_metrics = Channel
+    .fromFilePairs("$root/**/metrics/*.nii.gz",
+                    size: -1,
+                    maxDepth:2) {it.parent.parent.name}
+
+in_dwi_data = Channel
+    .fromFilePairs("$root/**/{*dwi.bval,*dwi.bvec,*dwi.nii.gz,*peaks.nii.gz}",
+                    size: 4,
+                    maxDepth:1,
+                    flat: true) {it.parent.name}
+
+(data_for_kernels, data_for_commit) = in_dwi_data
+    .map{sid, bval, bvec, dwi, peaks -> 
+        [tuple(sid, bval, bvec, dwi, peaks),
+        tuple(sid, bval, bvec, dwi, peaks)]}
+    .separate(2)
+
+
+in_t1_labels
+    .join(in_transfo)
+    .set{anat_for_transformation}
+process Transform_T1_Labels {
+    cpus 1
+
+    input:
+    set sid, file(anat), file(labels), file(mat), file(warp) from anat_for_transformation
+
+    output:
+    set sid, "${sid}__labels_warped_int16.nii.gz" into labels_for_transformation, labels_for_decompose
+    set sid, "${sid}__t1_warped.nii.gz" into anat_for_registration, anat_for_concatenate, anat_for_metrics
+
+    script:
+    """
+    antsApplyTransforms -d 3 -i $anat -r $warp -o "${sid}__t1_warped.nii.gz" -t $warp $mat -n Linear
+    antsApplyTransforms -d 3 -i $labels -r $warp -o labels_warped.nii.gz -t $warp $mat -n NearestNeighbor
+    scil_image_math.py convert labels_warped.nii.gz "${sid}__labels_warped_int16.nii.gz" --data_type int16
+    """
+}
+
+
+in_tracking
+    .into{tracking_for_decompose;tracking_for_kernel}
+
+tracking_for_decompose
+    .join(anat_for_concatenate)
+    .join(labels_for_decompose)
+    .set{tracking_labels_for_decompose}
+
+process Decompose_Connectivity {
+    cpus 2
+
+    input:
+    set sid, file(trackings), file(anat), file(labels) from tracking_labels_for_decompose
+
+    output:
+    set sid, "${sid}__decompose.h5" into h5_for_commit, h5_for_skip_commit
+
+    script:
+    """
+    no_pruning_arg=""
```

Comment:
These 3 first if could be in groovy like in pft_tracking options in tractoflow. Could you change that.